### ISSUE TITLE
Build instructions are out of date for Go 1.16 / go modules

### ISSUE THREAD

**Issue by gdbelvin** [keybase/client#24479]
Attempting to follow the build instructions produces the following:
```
go version
go version go1.16.2 darwin/amd64

go get github.com/keybase/client/go/keybase
go: downloading bazil.org/fuse v0.0.0-20200524192727-fb710f7dfd05
go: downloading github.com/go-errors/errors v1.1.1
go: downloading github.com/golang/mock v1.5.0
go: downloading github.com/miekg/dns v1.1.41
go: downloading golang.org/x/image v0.0.0-20210220032944-ac19c3e999fb
go: downloading golang.org/x/text v0.3.5
go: downloading google.golang.org/appengine v1.6.1
go: downloading github.com/asaskevich/govalidator v0.0.0-20180319081651-7d2e70ef918f
go: downloading github.com/lib/pq v1.2.0
go: downloading github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db
go: downloading github.com/sirupsen/logrus v1.4.1
go: downloading github.com/konsorten/go-windows-terminal-sequences v1.0.1
go: downloading github.com/golang/protobuf v1.3.1
go: downloading github.com/shopspring/decimal v1.2.0
go: downloading github.com/golang/protobuf v1.3.3
github.com/keybase/client/go/keybase imports
	github.com/keybase/client/go/service imports
	github.com/stellar/go/build: cannot find module providing package github.com/stellar/go/build
github.com/keybase/client/go/keybase imports
	github.com/keybase/client/go/service imports
	github.com/stellar/go/clients/horizon: cannot find module providing package github.com/stellar/go/clients/horizon

go install -tags production github.com/keybase/client/go/keybase
go install: version is required when current directory is not in a module
	Try 'go install github.com/keybase/client/go/keybase@latest' to install the latest version
```

Attempting to resolve the build dependencies myself...

```
mkdir -p ~/go/src/github.com/keybase
cd ~/go/src/github.com/keybase
git clone https://github.com/keybase/client.git
cd client/go
make gen-deps
./gen_deps.pl
go: cannot find main module, but found vendor/vendor.json in /Users/gdb/go/src/github.com/keybase/client/go
	to create a module there, run:
	go mod init
go: cannot find main module, but found vendor/vendor.json in /Users/gdb/go/src/github.com/keybase/client/go
	to create a module there, run:
	go mod init
total packages for which to calculate dependencies: 0
```

**Comment by kf5grd** [comment]
This is due to changes in Go 1.16. A quick workaround is to set the environment variable `GO111MODULE` to `auto`. You can set it permanently with `go env -w GO111MODULE=auto`.

See https://blog.golang.org/go116-module-changes

**Comment by gdbelvin** [comment]
Thanks. Any word on upgrading to go modules? 

I've tried doing a local upgrade myself but found that the version of stellar currently referenced in vendors.json stellar/e43266cae98f292a11218decae91a277a8489da2 appears to be in a deleted branch. 

```
git branch -a --contains e43266cae98f292a11218decae91a277a8489da2
error: no such commit e43266cae98f292a11218decae91a277a8489da2
```

Furthermore, the old version of Steller references other, deleted repos.

```
go get github.com/stellar/go@e43266cae98f292a11218decae91a277a8489da2
go: github.com/stellar/go@v0.0.0-20191010205648-0fc3bfe3dfa7 requires
	bitbucket.org/ww/goautoneg@v0.0.0-20120707110453-75cd24fc2f2c: reading https://api.bitbucket.org/2.0/repositories/ww/goautoneg?fields=scm: 404 Not Found
```

This is the root cause of the above build errors.

Any chance the stellar dependencies could be updated?

**Comment by kennethlakin** [comment]
+1 to this bug... it also affects me. I'm running `go` 1.17.2, and have tried the following:

```
go get github.com/keybase/client/go/keybase@v5.8.1
go get github.com/keybase/client/go/keybase@v5.8.0
go get github.com/keybase/client/go/keybase@v5.7.1
```

which all fail with

```
github.com/keybase/client/go/keybase imports
	github.com/keybase/client/go/service imports
	github.com/stellar/go/build: cannot find module providing package github.com/stellar/go/build
github.com/keybase/client/go/keybase imports
	github.com/keybase/client/go/service imports
	github.com/stellar/go/clients/horizon: cannot find module providing package github.com/stellar/go/clients/horizon
```

doing `go get github.com/keybase/client/go/keybase` (which -I assume- builds the latest in `master`) fails with that `bitbucket` 404 that @gdbelvin mentioned:

```
go: github.com/stellar/go@v0.0.0-20191010205648-0fc3bfe3dfa7 requires
	bitbucket.org/ww/goautoneg@v0.0.0-20120707110453-75cd24fc2f2c: reading https://api.bitbucket.org/2.0/repositories/ww/goautoneg?fields=scm: 404 Not Found
```

Just how the heck are official Keybase staff successfully building the Keybase binaries, and why do the publicly-accessible build instructions seem to be instructions that don't work?

**Comment by kennethlakin** [comment]
@gdbelvin So, I got "inspired" by  the instructions in this PR that was merged two weeks ago: <https://github.com/keybase/client/pull/24023>, and discovered that the build instructions for the latest commit in the `master` branch build Keybase just fine.

So, if you follow the instructions here: <https://github.com/keybase/client/tree/master/go#building>, if you have the same luck that I did, you will get a `keybase` executable.

If you _don't_, do know that I'm running `go` `1.17.2`, so maybe the code won't build with 1.16?

And, if `keybase` builds for you and you want to build `kbfsfuse` and the `git-remote-keybase` helpers, this did the trick for me:

```
go install -tags production github.com/keybase/client/go/kbfs/kbfsfuse
go install -tags production github.com/keybase/client/go/kbfs/kbfsgit/git-remote-keybase
```

I _do_ wonder how the `v5.8.1` release was built if the thing it depends on vanished in August, and the tag was made and the release was published in September...
